datadog/threadstats/base.py

```diff
@@ -13,7 +13,10 @@
 # stdlib
 from contextlib import contextmanager
 from functools import wraps
-from time import time
+try:
+    from time import monotonic as time  # type: ignore[attr-defined]
```

Comment:
That's kinda problematic here, since the time is used in several places to compute the actual timestamp of a metric, and not in a difference of `end - start` 

You'd need to separate the usages here

Reply:
I totally missed that. Will update.